File: test7Go/oop/models/pizza/VegetarianPizza.go
package pizza

import (
	"fmt"
)

type VegetarianPizza struct {
	SimplePizza
}

func NewVegetarianPizza() VegetarianPizza {
	simplePizza := NewSimplePizza("Vegetarian Pizza", 25.47)
	simplePizza.Topping = fmt.Sprintf("%s,tomato,mushroom", simplePizza.Topping)

	return VegetarianPizza{simplePizza}
}

File: test7C#/test7/Model/Pizza/VegetarianPizza.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace test7.Model.Pizza
{
    class VegetarianPizza : SimplePizza
    {
        public VegetarianPizza() : base("Vegetarian Pizza", Convert.ToDecimal(25.47))
        {
            base.Topping(base.Topping() + ",tomato,mushroom");
        }
    }
}
